index.html

```diff
@@ -1183,6 +1183,110 @@
+        their pattern of interaction with that page. However, the page does this by using
+        the same events it would use to make the page interactive, so we can't block this
+        information access either.
+
+The following subsections discuss how to review an API proposal that exposes data that
+provides a new way to infer each of the above categories of information. They explain how
+to <a data-cite="design-principles#leave-the-web-better">leave the web better than you
+found it</a>.
+
+### Handling acceptable information {#acceptable-information}
+
+<div class="practice">
+  <span class="practicelab">New APIs can add additional ways of getting acceptable
+  information that are guarded at least as strongly as the old ways.</span>
+</div>
+
+Acceptable information exposure is always qualified by the (possibly empty) set of
+user-controlled settings or permissions that guard access to it. For example, the URLs of
+resources, the timing of link clicks, and the referrer chain within a single origin are
+not guarded by anything; the scroll position is guarded by the setting to turn off
+javascript; and access to the camera or geolocation are guarded by permission prompts.
+
+Information that would be acceptable to expose under one set of guards might be
+unacceptable under another set, so when an API designer intends to explain that their new
+API is acceptable because an existing acceptable API already exposes the same information,
+they must be careful to ensure that their new API is only available under a set of guards
+that's at least as strict. Without those guards, they need to make the argument from
+scratch, without relying on the existing API.
+
+### Handling information that's being removed {#unacceptable-information}
+
+<div class="practice">
+  <span class="practicelab">If existing APIs provide access to some information, but
+  we have a plan to change those APIs to prevent that access, new APIs must not be added
+  that provide that same information without extra guards that make the information
+  acceptable.</span>
+</div>
+
+### Handling information we can't completely block {#sketchy-information}
+
+<div class="practice">
+  <span class="practicelab">New APIs that provide access to undesirable information should
+  not make that information easier to access.</span>
+</div>
+
+
+1. If future web platform changes make it possible to remove other access to the
+   undesirable information, it should be clear how to extend those changes to the proposed
+   API.
+
+1. If an existing browser does block access to the undesirable information, perhaps by
+   breaking some experiences on the Web that other browsers don't wish to break, it should
+   be clear how the more-private browser can also prevent the new API from exposing that
+   information.
+
+1. A new API should be harder to use to discover undesirable information than the existing
+   APIs, for example by requiring more code, more maintenance, and/or more runtime cost.
```

Comment:
I think this is a good idea but I don't think the examples are particularly fitting, which confused me for a second. In my view we don't want to design APIs that, say, enable only dedicated attackers that will have no issue sending over 100KB of JS that spends 10 seconds to compute joint identities. We want to make attacks harder in ways that can't be scaled with computing and engineering resources, such as increasing the site usage time needed until sufficient information can be gathered from a single user, increasing user effort and user exposure or by requiring increasingly tight cooperation with other (malicious) parties.

Reply:
The situation in this section is that we have an existing API that lets attackers send 100KB of JS and spend 10s, and someone's proposing a new API that still exposes the information, but needs 1MB of JS or 100s. Sure it'd be nice to make the existing API more difficult too, but the other part of this section is that nobody has figured out how to do that.

Clearly I didn't explain that well ... any ideas how to do it better?

Reply:
Hmm do you think that the computational effort really makes a difference in favor of the new API here? I don't really see why it should (and to be honest I'm also struggling to come up with an example where we would encounter this kind of situation in practice).

Overall if the first two sub-points here are adhered to I don't see why an API would have to be additionally strictly more computationally expensive (and how that would even be measured).

Reply:
I think you're right that it just has to be not-easier, so I've changed the text in that direction. For concrete examples, `<a ping>` clearly reduced the cost of tracking clicks, since it removed the round trips from going through a redirect. Vs using a screencast API to detect adblocking, which is more difficult than looking for deleted DOM nodes or missing server-side requests.